epix/run_epix.py

```diff
@@ -0,0 +1,202 @@
+        tnow = monitor_time(tnow, 'cluster finding.')
+
+    result = epix.cluster(inter, args['tag_cluster_by'] == 'energy')
+
+    if args['debug']:
+        tnow = monitor_time(tnow, 'cluster merging.')
+
+    # Add eventid again:
+    result['evtid'] = ak.broadcast_arrays(inter['evtid'][:, 0], result['ed'])[0]
+
+    # Sort detector volumes and keep interactions in selected ones:
+    if args['debug']:
+        print('Removing clusters not in volumes:', *[v.name for v in args['detector_config']])
+        print(f'Number of clusters before: {np.sum(ak.num(result["ed"]))}')
+
+    # Returns all interactions which are inside in one of the volumes,
+    # Checks for volume overlap, assigns Xe density and create S2 to
+    # interactions. EField comes later since interpolated maps cannot be
+    # called inside numba functions.
+    res_det = epix.in_sensitive_volume(result, args['detector_config'])
+
+    # Adding new fields to result:
+    for field in res_det.fields:
+        result[field] = res_det[field]
+    m = result['vol_id'] > 0  # All volumes have an id larger zero
+    result = result[m]
+
+    # Removing now empty events as a result of the selection above:
+    m = ak.num(result['ed']) > 0
+    result = result[m]
+
+    if args['debug']:
+        print(f'Number of clusters after: {np.sum(ak.num(result["ed"]))}')
+        print('Assigning electric field to clusters')
+
+    if not ak.any(m):
+        # There are not any events left so return empty array:
+        warnings.warn('No interactions left, return empty DataFrame.')
+        if return_df:
+            if args.output_path and not os.path.isdir(args.output_path):
+                os.makedirs(args.output_path)
+
+            output_path_and_name = os.path.join(args.output_path,
+                                                args['file_name'][:-5] + "_wfsim_instructions.csv")
+            df = pd.DataFrame()
+            df.to_csv(output_path_and_name, index=False)
+            return
+
+    # Add electric field to array:
+    efields = np.zeros(np.sum(ak.num(result)), np.float32)
+    # Loop over volume and assign values:
+    for volume in args['detector_config']:
+        if isinstance(volume.electric_field, (float, int)):
+            ids = epix.awkward_to_flat_numpy(result['vol_id'])
+            m = ids == volume.volume_id
+            efields[m] = volume.electric_field
+        else:
+            efields = volume.electric_field(epix.awkward_to_flat_numpy(result.x),
+                                            epix.awkward_to_flat_numpy(result.y),
+                                            epix.awkward_to_flat_numpy(result.z)
+                                            )
+
+    result['e_field'] = epix.reshape_awkward(efields, ak.num(result))
+
+    # Sort in time and set first cluster to t=0, then chop all delayed
+    # events which are too far away from the rest.
+    # (This is a requirement of WFSim)
+    result = result[ak.argsort(result['t'])]
+
```

Comment:
The comments says delayed interactions are cut. This happens not at this stage of the code. -> Update the comment?

Reply:
Time of first cluster to zero is also deprecated, isn't it?

Reply:
I think the cutting is missing right now. As I understood we agreed on cutting all interactions which arrive later then 10**7 ns after the first interaction, did not we?

Reply:
Yes, interactions delayed by more than `max_delay` should be cut. The default value for `max_delay` was/is 1e7 I think. By the way the cutting is included in this [line](https://github.com/XENONnT/epix/blob/strax_interface/epix/run_epix.py#L128) but only for `source_rate == -1` case.